aderyn_core/src/detect/high/deletion_nested_mapping.rs

```diff
@@ -0,0 +1,135 @@
+        for delete_operation in context
+            .unary_operations()
+            .into_iter()
+            .filter(|op| op.operator == "delete")
+        {
+            if let Expression::IndexAccess(IndexAccess {
+                base_expression, ..
+            }) = delete_operation.sub_expression.as_ref()
+            {
+                if let Expression::Identifier(Identifier {
+                    referenced_declaration: Some(referenced_id),
+                    type_descriptions,
+                    ..
+                }) = base_expression.as_ref()
+                {
+                    // Check if we're deleting a value from mapping
+                    if type_descriptions
+                        .type_string
+                        .as_ref()
+                        .is_some_and(|type_string| type_string.starts_with("mapping"))
+                    {
+                        // Check if the value in the mapping is of type struct that has a member which is also a mapping
+                        if let Some(ASTNode::VariableDeclaration(VariableDeclaration {
+                            type_name: Some(TypeName::Mapping(Mapping { value_type, .. })),
+                            ..
+                        })) = context.nodes.get(referenced_id)
+                        {
+                            if let TypeName::UserDefinedTypeName(UserDefinedTypeName {
+                                referenced_declaration,
+                                ..
+                            }) = value_type.as_ref()
+                            {
+                                if let Some(ASTNode::StructDefinition(structure)) =
+                                    context.nodes.get(referenced_declaration)
+                                {
+                                    // Check that a member of a struct is of type mapping
+                                    if structure.members.iter().any(|member| {
+                                        member.type_descriptions.type_string.as_ref().is_some_and(
+                                            |type_string| type_string.starts_with("mapping"),
+                                        )
+                                    }) {
+                                        capture!(self, context, delete_operation);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::High
+    }
+
+    fn title(&self) -> String {
+        String::from("Deletion from a nested mappping.")
+    }
+
+    fn description(&self) -> String {
+        String::from("A deletion in a structure containing a mapping will not delete the mapping. The remaining data may be used to compromise the contract.")
```

Comment:
"as-is" from Slither for parity

Reply:
Can you help me understand the recommendation? I didn't quite get it : (

Reply:
Good question @TilakMaddy , it's not clear. My first thought was to just add a boolean `locked` field in the struct that you can check when retrieving the balances, but that would not prevent direct mapping references.

I'm not actually dure